### PR TITLE
Configuration avoidance API and branch name inference

### PR DESCRIPTION
**Pull request by Joaquimmnetto**
## Description
Added configuration avoidance API use, as it is necessary for the project itself to not require github connection with the new implementation. It also has the added bonus of making gradle project synching faster. 

Besides this, now we are using the new `atlas-github` version for git branch name inference. It first tries to get a property (`github.branch.name`), and if it does not exists, it goes with the following: 

* if the current branch name starts with "PR-" (eg. PR-145), it fetches the PR branch name using github API. 
* else, it returns the current local branch tracking branch name (as got by grgit)
* else, it returns the current local branch name (as got by grgit)

## Changes
* ![IMPROVE] refactored PluginsPlugin class to use gradle's new configuration avoidance API
* ![ADD] branch name inference and improved RepositoryInfo object
* ![ADD] `sonar.branch.name`  sonarqube property

[NEW]:https://atlas-resources.wooga.com/icons/icon_new.svg "New"
[ADD]:https://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:https://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:https://atlas-resources.wooga.com/icons/icon_change.svg "Change"
[FIX]:https://atlas-resources.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:https://atlas-resources.wooga.com/icons/icon_update.svg "Update"

[BREAK]:https://atlas-resources.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:https://atlas-resources.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:https://atlas-resources.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:https://atlas-resources.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:https://atlas-resources.wooga.com/icons/icon_webGL.svg "WebGL"
[UNITY]:https://atlas-resources.wooga.com/icons/icon_unity.svg "Unity"
[GRADLE]:https://atlas-resources.wooga.com/icons/icon_gradle.svg "Gradle"
[LINUX]:https://atlas-resources.wooga.com/icons/icon_linux.svg "Linux"

### CODE REVIEW

**Comment by Larusso** [comment]
Would the patches to the github plugin alter this PR? Shall I review it as is?

**Comment by Joaquimmnetto** [comment]
@Larusso you can review this now :)

File: src/main/groovy/wooga/gradle/plugins/PluginsPlugin.groovy [hunk 1]
```diff
@@ -176,149 +178,145 @@ class PluginsPlugin implements Plugin<Project> {
...
         def publishTaskProvider = tasks.named(GithubPublishPlugin.PUBLISH_TASK_NAME)
         publishTaskProvider.configure {GithubPublish githubPublishTask ->
             githubPublishTask.onlyIf(new ProjectStatusTaskSpec("rc", "final"))
             githubPublishTask.with {
                 releaseName.set(project.provider {project.version.toString()})
                 tagName.set(project.provider {"v${project.version}"})
-                targetCommitish.set(project.extensions.grgit.branch.current.name as String)
                 prerelease.set(project.properties['release.stage']!='final')
-                body.set(releaseNotesTask.output.map{it.asFile.text })
+                body.set(releaseNotesTask.get().output.map{it.asFile.text })
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/plugins/PluginsPlugin.groovy:251]:
```suggestion
                body.set(releaseNotesTask.flatMap({it.output.map({it.asFile.text })})
```
We should use flatMap here to not call `get`. I know that you are in a configure block so it should be already lazy.

**Review by Larusso — APPROVED** [review]
Ok looks good to me except the one optional issue with flatmap instead of calling get.
It is practically impossible with this complex interaction of multiple tasks/plugins to verify that the configuration avoidance is used at its max. But it looks good enough I would say.

File: src/main/groovy/wooga/gradle/plugins/PluginsPlugin.groovy (hunk 1 above)
Comment (Joaquimmnetto) [reply src/main/groovy/wooga/gradle/plugins/PluginsPlugin.groovy:264]: Yeah, but it is still better to pass the provider, I'm aware. I'll accept the change.

**Review by Larusso — APPROVED** [review]

**Pull request merged**